docs/docs/python/integrations/aws-lambda.mdx

```diff
@@ -66,6 +65,40 @@ Use [layout](../../../reference/targets/python_aws_lambda_function.mdx#layout) t
 `file` / `files` targets will not be included in the built AWS Lambda artifacts because filesystem APIs like `open()` would not load them as expected. Instead, use the `resource` and `resources` target. See [Assets and archives](../../using-pants/assets-and-archives.mdx) for further explanation.
 :::
 
+### Specifying a runtime explicitly
+
+Pants will attempt to infer the appropriate runtime and its platform-specific details from [the relevant interpreter constraints](../overview/interpreter-compatibility.mdx). For example, given `interpreter_constaints = ["==3.12.*"]`, Pants will build the Lambda package to appropriate for the  `python3.12` runtime. The Lambda environment is a relatively stable, running Linux on either a x86-64 or arm64 CPU, and thus Pants includes default "complete platforms" specifications that allow choosing appropriate artifacts for third-party dependencies with native code. These are chosen based on the (inferred or explicit) runtime and [the architecture](../../../reference/targets/python_aws_lambda_function.mdx#architecture).
+
+The inference may not work in your scenario and/or may not be appropriate:
+
+1. if the interpreter constraints are not obviously referring to all versions from a given minor release series, Pants won't infer the runtime. For instance, `>=3.11,<3.13` is too wide (it covers both 3.11 and 3.12), while `==3.12.1` is too specific (AWS's `python3.12` runtime may not use that patch version).
+2. if the interpreter constraints refer to a runtime that Pants doesn't have a default complete platforms JSON specification for, Pants will require you to specify one.
+3. if the cloud environment has changed, Pants' default complete platforms value may imply correct artifacts dependency and you may want to customise.
+
+In the first case, [the `runtime` field](../../../reference/targets/python_aws_lambda_function.mdx#runtime) can be used to specify which runtime is intended for each `python_aws_lambda_function` and `python_aws_lambda_layer` target. It should be one of the values from [https://docs.aws.amazon.com/lambda/latest/dg/lambda-python.html](https://docs.aws.amazon.com/lambda/latest/dg/lambda-python.html).
```

Comment:
This makes sense! I would just move this before the part talking about inferring runtime from interpreter constraints.